types/inquirer/index.d.ts

```diff
@@ -10,430 +10,909 @@
-         * Register the defaults provider prompts
+         * The type of the choice.
          */
-        restoreDefaultPrompts(): void;
+        type?: string;
     }
 
-    interface QuestionCommon<A> {
+    /**
+     * Provides options for a choice.
+     *
+     * @template T
+     * The type of the answers.
+     */
+    interface ChoiceOptions<T extends Answers = Answers> extends ChoiceBase {
+        /**
+         * @inheritdoc
+         */
+        type?: "choice";
+
         /**
-         * The name to use when storing the answer in the answers hash.
-         * If the name contains periods, it will define a path in the answers hash.
+         * The name of the choice to show to the user.
          */
         name?: string;
+
         /**
-         * The question to print. If defined as a function, the first parameter will be
-         * the current inquirer session answers.
-         * Defaults to the value of `name` (followed by a colon).
+         * The value of the choice.
          */
-        message?: string | ((answers: A) => string);
+        value?: any;
+
         /**
-         * Default value(s) to use if nothing is entered, or a function that returns
-         * the default value(s). If defined as a function, the first parameter will be
-         * the current inquirer session answers.
+         * The short form of the name of the choice.
          */
-        default?: string | number | boolean | any[] | ((answers: A) => any) | ((answers: A) => Promise<any>);
+        short?: string;
+
         /**
-         * Change the default _prefix_ message.
+         * The extra properties of the choice.
          */
-        prefix?: string;
+        extra?: any;
+    }
+
+    /**
+     * Provides options for a choice of the `ListPrompt`.
+     *
+     * @template T
+     * The type of the answers.
+     */
+    interface ListChoiceOptions<T extends Answers = Answers> extends ChoiceOptions<T> {
         /**
-         * Change the default _suffix_ message.
+         * A value indicating whether the choice is disabled.
          */
-        suffix?: string;
+        disabled?: DynamicQuestionProperty<boolean | string, T>;
+    }
+
+    /**
+     * Provides options for a choice of the `CheckboxPrompt`.
+     *
+     * @template T
+     * The type of the answers.
+     */
+    interface CheckboxChoiceOptions<T extends Answers = Answers> extends ListChoiceOptions<T> {
         /**
-         * Receive the current user answers hash and should return `true` or `false` depending
-         * on whether or not this question should be asked. The value can also be a simple boolean.
+         * A value indicating whether the choice should be initially checked.
          */
-        when?: boolean | ((answers: A) => boolean | Promise<boolean>);
+        checked?: boolean;
     }
 
-    interface QuestionOptions<A> {
+    /**
+     * Provides options for a choice of the `ExpandPrompt`.
+     *
+     * @template T
+     * The type of the answers.
```

Comment:
This might be a good idea.
I tried to do some coding similar to the ".NET"-library where all generic parameters are named `T` expect there are multiple generic types (like, for example, `List<T>` and `Dictionary<TKey, TValue>`).

If you want me to change the names of the generic parameters to `TAnswers` rather than `T` I'm, of course, willing to do so.

Reply:
If you're not going to remove the comment, then `T` is just fine.

Reply:
Alright!
Let's keep it as it is then.